test-sbt/shared/src/main/scala/zio/test/sbt/BaseTestTask.scala

```diff
@@ -18,7 +18,7 @@ abstract class BaseTestTask(val taskDef: TaskDef, testClassLoader: ClassLoader)
 
   protected def run(eventHandler: EventHandler, loggers: Array[Logger]) =
     for {
-      res    <- spec.run.provide(new SbtTestLogger(loggers) with Clock.Live)
+      res    <- spec.run(args).provide(new SbtTestLogger(loggers) with Clock.Live)
```

Comment:
Can we have the SBT test runner filter the spec here and then run the resulting spec? I think that would make this more modular where the testing framework is exposed methods like `filter` and `run` and then the client, in this case the test runner, can combine those to get the desired output instead of baking the concept of `run` taking a certain kind of command line parameter.

Reply:
Hmm, I haven't found a great way to do this from `BaseTestTask`. I guess I could filter there and change `AbstractRunnableSpec.run` to accept the new spec and use that instead, but I'm not sure if that's what you were going for. Maybe I'm just not seeing the better solution. Because I can't swap out the spec in `AbstractRunnableSpec` (it being abstract and all).

Reply:
You can do something like:

```scala
      res <- args.testSearchTerm
              .fold(spec.run)(
                s => spec.runner.run(spec.spec.filterTestLabels(_.toString.contains(s)).getOrElse(spec.spec))
              )
              .provide(new SbtTestLogger(loggers) with Clock.Live)
```

May make sense to pull the filtering logic out into a separate helper function.